numba/tests/test_try_except.py

```diff
@@ -208,6 +208,165 @@ def udt(x):
         self.assertEqual(res, 123)
 
 
+    def test_recursion(self):
+        @njit
+        def foo(x):
+            if x > 0:
+                try:
+                    foo(x - 1)
+                except:
+                    print("CAUGHT")
+                    return 12
+            if x == 1:
+                raise ValueError("exception")
+
+        with captured_stdout() as stdout:
+            res = foo(10)
+
+        self.assertEqual(stdout.getvalue().split(), ["CAUGHT",],)
+
+    # this is broken
```

Comment:
This comment seems useless, makes me ask the question, what exactly is broken. And maybe it isn't broken anymore.

Reply:
Oups, wrong window, nevermind.

Reply:
oops, that's left over comments from tests written by @stuartarchibald. I thought I removed them all.